### PR TITLE
Filters

### PR DESCRIPTION
**Pull request by theprogrammerinyou**
Added Category filters, price filters and rating filters, only UI not the functionality.

The improvement to products commit was not intended to be pushed in this branch, but please do review that as well. 
Thank you.

### CODE REVIEW

**Review by GuptaEesh — COMMENTED** [review]
Good use of named exports.
Good use of variables for styling.
Good to merge.

File: src/components/Filters/CategoryFilters.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import { Grid, Checkbox, Typography, FormControlLabel } from "@mui/material";
+import React, { useReducer } from "react";
```
Comment (junaidshaikh-js) [inline src/components/Filters/CategoryFilters.jsx:2]: No need to import React from version 17

**Review by junaidshaikh-js — APPROVED** [review]
Great work! 

I liked how you separated each component and its logic into separate files. 

Can be merged.

File: src/components/Filters/CategoryFilters.jsx (hunk 1 above)
Comment (theprogrammerinyou) [reply src/components/Filters/CategoryFilters.jsx:2]: Thanks for this, was looking for a way to stop importing thus tried using eslint rule, but this helps. Thank you.

**Pull request merged**